pyfr/integrators/dual/pseudo/pseudocontrollers.py

```diff
@@ -140,8 +145,27 @@ def __init__(self, *args, **kwargs):
                     )
                 )
 
+        self.pseudo_step_multiplied(1.0)
+
         self.backend.commit()
 
+    def adjust_pseudo_step(self, dt):
+        old_dtau = self.dtau
+        super().adjust_pseudo_step(dt)
+        self.pseudo_step_multiplied(self.dtau / old_dtau)
+
+    def pseudo_step_multiplied(self, y):
```

Comment:
Maybe make this a function that purely does the binding of dtau_min, dtau_max and dtau_fieldf and move the `dtau_min *= y` and `self.dtau_max *= y`. This should make the calls where you do `pseudo_step_multiplied` more understandable.